packages/generator/src/options/options-parser.ts

```diff
@@ -1,12 +1,62 @@
 import { createLogger } from '@sap-cloud-sdk/util';
-import { Options } from 'yargs';
-import { ParsedOptions } from './generator-options';
+import { InferredOptionType, Options } from 'yargs';
 const logger = createLogger('generator-options');
 
-type Option = Options & {
+/**
+ * @internal
+ * Options for SAP Cloud SDK generators.
+ * Some keys are modified, for more explicit parsing.
+ */
+export type Option = Omit<Options, 'coerce'> & {
+  /**
+   * Name of the option the current option was replaced with.
+   */
   replacedBy?: string;
-  describe: string;
-};
+
+  /**
+   * Only required options should set `demandOption`. Other options are not required by default.
+   */
+  demandOption?: true;
+
+  /**
+   * Coerce function with additional parameter for all options.
+   */
+  coerce?: (arg: any, options: any) => any;
+} & Required<Pick<Options, 'describe' | 'type'>>;
+
+/**
+ * @internal
+ * Helper to represent parsed options based on a public generator options type and a CLI options configuration.
+ * - Makes all properties required.
+ * - Removes deprecated and replaced options.
+ * - Replaces input types with coerced types.
```

Comment:
[pp] Mention that this means that for example path are resolved.

Reply:
But it doesn't, it depends what the `coerce` function does. I will add it to the path arg coersion functions instead.